llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgTransform/StructuredTransformOpsExt.h

```diff
@@ -57,6 +57,21 @@ class TrackingListener : public RewriterBase::Listener,
 #endif // LLVM_ENABLE_ABI_BREAKING_CHECKS
   }
 
+  DiagnosedSilenceableFailure check(Location loc) {
+    if (failed(checkErrorState()))
+      return emitDefiniteFailure(loc, "listener failed");
+    return DiagnosedSilenceableFailure::success();
+  }
+  DiagnosedSilenceableFailure check(Location loc,
+                                    DiagnosedSilenceableFailure &&diag) {
```

Comment:
Note that the error message is reported when _constructing_ the definite failure, taking an rvalue doesn't do anything to that here. So the error message will be printed even if listenerState is a success.